### DISCUSSION TITLE
Making entry point names of `aiida-core` consistent with our own naming conventions

### DISCUSSION THREAD

**Discussion by sphuber** [aiidateam/aiida-core#5099] [Ideas] [answered]
Hello everyone :wave: . As you know, we have an unwritten naming convention for entry points names in that they should be prefixed with a name specific to the plugin package, typically the name of the package. For example, the `aiida-quantumespresso` plugin package prefixes all its entry points with `quantumespresso.`. This approach has two major benefits:

1. It makes it directly clear from the entry point from what plugin package it comes from.
2. It namespaces all entry points making it unlikely that the entry points of different packages will clash. If two or more entry points have the exact same entry point name, they will not be able to be loaded.

The AiiDA plugin registry has some checks to verify these unwritten rules, but so far they cannot be really enforced. A big part of this is due to the fact that `aiida-core` itself does not respect these guidelines. With the exception of one, none of its entry points are properly prefixed with `core.`. I am of the opinion that it would be beneficial, especially in the long run, to correct this. It will make it much easier to know that certain plugins come from `aiida-core`, but also, we are no longer hijacking certain namespaces that could be used for plugin packages instead. For example, it would be difficult to create the hypothetical `aiida-array` package now since we have unrightfully claimed the `array` namespace in the `aiida.data` entry point group.

The only downside that I see to making this change is backwards compatibility. However, as shown by [this PR](https://github.com/aiidateam/aiida-core/pull/5073), which already implements the changes, it is possible to do in a backwards compatible way. It also, as usual, provides full automated data migrations to make existing data fully consistent (the entry point strings are stored in the database in certain cases). The old entry point strings will still work when passed to the various factories, for example:
```python
ArrayData = DataFactory('array')
```
will still work exactly the same, except it will emit a deprecation warning to change it to:
```python
ArrayData = DataFactory('core.array')
```
As you can see, also adapting existing code to this change is easy as this is easily search-and-replaceable. The same goes for the command line. The entry point names will continue to work there with a deprecation warning, giving people an easy way to update their workflow. For example,
```console
verdi node rehash --entry-point int
```
will still work and is equivalent to the modern:
```console
verdi node rehash --entry-point core.int
```
Since we are getting close to the `v2.0` release, we would like to include this change. Still, we wanted to quickly gauge your opinion on this proposed change.

**Comment by chrisjsewell** [comment, marked as answer]
This has now been closed in #5073
